### PR TITLE
[Typing][PEP585 Upgrade][BUAA][61-70] Use standard collections for type hints for 10 files in `python/paddle/`

### PR DESCRIPTION
**Pull request by Fripping**
### PR Category
User Experience

### PR Types
Devs

### Description

为如下文件添加PEP563，并进行PEP585升级：
- python/paddle/incubate/autograd/primapi.py
- python/paddle/incubate/autograd/primx.py
- python/paddle/incubate/framework/random.py
- python/paddle/incubate/nn/attn_bias.py
- python/paddle/io/dataloader/batch_sampler.py
- python/paddle/io/dataloader/dataset.py
- python/paddle/io/dataloader/sampler.py
- python/paddle/inference/wrapper.py
- python/paddle/io/reader.py
- python/paddle/jit/dy2static/ast_utils.py

### Related links
- #66936
@gouzil

### CODE REVIEW

File: python/paddle/jit/dy2static/ast_utils.py [hunk 1]
```diff
@@ -32,7 +32,7 @@ def ast_to_source_code(ast_node):
     if isinstance(ast_node, gast.AST):
         ast_node = gast.gast_to_ast(ast_node)
 
-    if sys.version_info >= (3, 9):
```
Comment (SigureMo) [inline python/paddle/jit/dy2static/ast_utils.py]: ？

File: python/paddle/incubate/nn/attn_bias.py [hunk 2]
```diff
@@ -57,7 +58,7 @@ def materialize(self, shape, dtype=paddle.float32):
 class SeqLenInfo:
     seqstart: paddle.Tensor
     max_seqlen: int
-    seqstart_py: List[int]
+    seqstart_py: list[int]
```
Comment (SigureMo) [inline python/paddle/incubate/nn/attn_bias.py:65]: 这都没用 PEP 563 就直接用？

File: python/paddle/incubate/nn/attn_bias.py [hunk 3]
```diff
@@ -21,8 +21,9 @@
 # LICENSE file in the root directory of this source tree.
 
 from abc import ABC, abstractmethod
+from collections.abc import Sequence
 from dataclasses import dataclass
-from typing import List, Optional, Sequence
+from typing import Optional
```
Comment (SigureMo) [inline python/paddle/incubate/nn/attn_bias.py]: Optional 不需要替换吗

Comment (Fripping) [reply python/paddle/incubate/nn/attn_bias.py]: 收到

File: python/paddle/incubate/framework/random.py [hunk 4]
```diff
@@ -14,13 +14,15 @@
 
 from __future__ import annotations
 
-from typing import TYPE_CHECKING, Literal, Protocol, Sequence, overload
+from typing import TYPE_CHECKING, Literal, Protocol, overload
```
Comment (gouzil) [inline python/paddle/incubate/framework/random.py]: `Literal, Protocol, Sequence` 这几个都可以在 `TYPE_CHECKING` 导入

File: python/paddle/io/dataloader/batch_sampler.py [hunk 5]
```diff
@@ -15,7 +15,7 @@
 from __future__ import annotations
 
 import math
-from typing import (
+from collections.abc import (
```
Comment (SigureMo) [inline python/paddle/io/dataloader/batch_sampler.py]:
```suggestion
from typing import (
```

运行时会用到，暂时不能替换

**Review by SigureMo — DISMISSED** [review]
LGTMeow

**Review by SigureMo — APPROVED** [review]

**Pull request merged**